presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -358,6 +376,67 @@ public void testCreateAcidTable(boolean isPartitioned, BucketingType bucketingTy
         }
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL)
+    public void testFilesForAbortedTransactionsIgnored()
+            throws TException
+    {
+        if (getHiveVersionMajor() < 3) {
+            throw new SkipException("Hive transactional tables are supported with Hive version 3 or above");
+        }
+
+        String tableName = "test_aborted_transaction_table";
+        onHive().executeQuery("" +
+                "CREATE TABLE " + tableName + " (col INT) " +
+                "STORED AS ORC " +
+                "TBLPROPERTIES ('transactional'='true')");
+
+        ThriftHiveMetastoreClient client = transactionalTestHelper.createMetastoreClient();
+        try {
+            String selectFromOnePartitionsSql = "SELECT col FROM " + tableName + " ORDER BY COL";
+
+            // Create `delta-A` file
+            onHive().executeQuery("INSERT INTO TABLE " + tableName + " VALUES (1),(2)");
+            QueryResult onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsExactly(row(1), row(2));
+
+            // Insert data to create a valid delta, which creates `delta-B`
+            onHive().executeQuery("INSERT INTO TABLE " + tableName + " SELECT 3");
+
+            // Simulate aborted transaction in Hive which has left behind a write directory and file
+            long transaction = client.openTransaction("test");
+
+            client.allocateTableWriteIds("default", tableName, Collections.singletonList(transaction)).get(0).getWriteId();
+
+            // Rollback transaction which leaves behind a delta directory `delta-C` i.e. `delta_0000003_0000003_0000`
+            client.abortTransaction(transaction);
+
+            String deltaA = warehouseDirectory + "/" + tableName + "/delta_0000001_0000001_0000/bucket_00000";
+            String deltaB = warehouseDirectory + "/" + tableName + "/delta_0000002_0000002_0000/bucket_00000";
+            String deltaC = warehouseDirectory + "/" + tableName + "/delta_0000003_0000003_0000/bucket_00000";
+
+            // Delete original `delta-B`, `delta-C`
+            hdfsClient.delete(deltaB);
+            hdfsClient.delete(deltaC);
+
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            hdfsClient.loadFile(deltaA, byteArrayOutputStream);
+
+            // Copy content of `delta-A` to `delta-B`
+            hdfsClient.saveFile(deltaB, new ByteArrayInputStream(byteArrayOutputStream.toByteArray()));
+
```

Comment:
```suggestion
// Verify that data from delta-A and delta-B is visible
assertThat(onePartitionQueryResult).containsOnly(row(1), row(1), row(2), row(2));
```

Reply:
fixed.

Reply:
I meant to add extra step with just delta-A and delta-B. The comment by the last assertion was fine. Sorry for not being clear.

Reply:
got it, fixed.